m2cgen/assemblers/boosting.py

```diff
@@ -5,36 +5,40 @@
 from m2cgen.assemblers.base import ModelAssembler
 
 
-class XGBoostModelAssembler(ModelAssembler):
+class BaseBoostAssembler(ModelAssembler):
 
-    def __init__(self, model):
-        super().__init__(model)
-        self._base_score = self.model.base_score
+    classifier_name = None
 
-        feature_names = self.model.get_booster().feature_names
-        self._feature_name_to_idx = {
-            name: idx for idx, name in enumerate(feature_names)
-        }
+    def __init__(self, model, trees, base_score=0):
+        super().__init__(model)
+        self.trees = trees
+        self._base_score = base_score
 
         self._output_size = 1
         self._is_classification = False
+
         model_class_name = type(model).__name__
-        if model_class_name == "XGBClassifier":
+        if model_class_name == self.classifier_name:
             self._is_classification = True
-            if self.model.n_classes_ > 2:
-                self._output_size = self.model.n_classes_
+            if model.n_classes_ > 2:
+                self._output_size = model.n_classes_
 
     def assemble(self):
-        model_dump = self.model.get_booster().get_dump(dump_format="json")
-        trees = [json.loads(d) for d in model_dump]
-
         if self._is_classification:
             if self._output_size == 1:
-                return self._assemble_bin_class_output(trees)
+                return self._assemble_bin_class_output(self.trees)
             else:
-                return self._assemble_multi_class_output(trees)
+                return self._assemble_multi_class_output(self.trees)
         else:
-            return self._assemble_single_output(trees, self._base_score)
+            return self._assemble_single_output(self.trees, self._base_score)
+
+    def _assemble_single_output(self, trees, base_score=0):
```

Comment:
We already have a `trees` attribute in `self` so it looks quite confusing to pass it in arguments as well. Can we rename the `self` one to `all_trees` or something?

Reply:
Done